pilot/test/mock/config.go

```diff
@@ -516,12 +519,22 @@ func CheckCacheEvents(store model.ConfigStore, cache model.ConfigStoreCache, nam
 	CheckMapInvariant(store, t, namespace, n)
 
 	log.Infof("Waiting till all events are received")
-	util.Eventually(func() bool {
-		lock.Lock()
-		defer lock.Unlock()
-		return added == n && deleted == n
-
-	}, t)
+	timeout := time.After(60 * time.Second)
+	for {
+		select {
+		case <-timeout:
+			t.Fatalf("timeout waiting to receive expected events. actual added %d deleted %d. expected %d",
+				added, deleted, n)
+		default:
+			lock.Lock()
+			if added == n && deleted == n {
+				lock.Unlock()
+				return
+			}
+			lock.Unlock()
```

Comment:
All the locks and conditionals can be removed by using an `added`, `removed` channel here